### PR TITLE
Move build to Hosted macOS and remove unnecessary LightGBM reference.

### PR DESCRIPTION
**Pull request by montebhoover**
After updating LightGBM, we can now move build to Hosted macOS.  Also removed unnecessary LightGBM reference in response to comment on #1234.

### CODE REVIEW

**Comment by eerhardt** [comment]
You'll need to `brew install` more things on the mac.  I think `libgdiplus` is one.

/cc @safern @Ivanidzo4ka

**Comment by safern** [comment]
> You'll need to brew install more things on the mac. I think libgdiplus is one.

Yes, we need to add a step to install dependencies in the mac leg. 

For `libgdiplus` you need to do:
`brew install libgdiplus`
`brew link gettext --force`

**Comment by montebhoover** [comment]
No problem, I'll update the docs here.

---
In reply to: [429396530](https://github.com/dotnet/machinelearning/pull/1240#issuecomment-429396530) [](ancestors = 429396530)

**Comment by montebhoover** [comment]
Thanks. Are these required at build time or run time?

---
In reply to: [429398100](https://github.com/dotnet/machinelearning/pull/1240#issuecomment-429398100) [](ancestors = 429398100)

**Comment by safern** [comment]
`libgdiplus` is a runtime requirement. It is used for image manipulation, there is a set of tests that need it, from what I know there are some image ML APIs in the ML framework, it uses .NET Core's `System.Drawing.Common` package which in Unix depends on `libgdiplus` being installed.

**Comment by montebhoover** [comment]
Okay, thanks! So libgdiplus and gettext are unrelated to the LightGBM change, but are requirements from new image transforms?  I am having trouble validating this locally.  The addition of the gdi lib caused my tensorflow tests to pass, but seems to cause new failures in kmeans tests (the metrics assertions are wildly incorrect now).

---
In reply to: [429446165](https://github.com/dotnet/machinelearning/pull/1240#issuecomment-429446165) [](ancestors = 429446165)

**Comment by eerhardt** [comment]
> Okay, thanks! So libgdiplus and gettext are unrelated to the LightGBM change, but are requirements from new image transforms?.... The addition of the gdi lib caused my previously failing tensorflow tests to pass

Yes, `libgdiplus` is necessary for any image transforms - which is why the tensorflow tests pass - they all use image transforms.

> but seems to cause new failures in kmeans tests (the metrics assertions are wildly incorrect now).

With your latest run, there are only 2 tests failing:

1. TimeSeries.ChangePointDetectionWithSeasonality()
```
Assert.Equal() Failure
Expected: 1.6069465 (rounded from 1.60694649815559)
Actual: 1.6069039 (rounded from 1.60690385103226)
```
2. NormalizerTests.LpGcNormAndWhiteningWorkout
```
System.InvalidOperationException : Invalid arguments to LAPACK gesvd, error: -6
```

I would start by debugging the 2nd test failure to see what is going wrong. The 1st test failure looks "close enough", though I wonder why moving from our private queue to a hosted queue would have changed the number slightly?

**Comment by montebhoover** [comment]
Just wanted to mention that I haven't been able to look at this for the past few days since I've been working on prepping the release of NimbusML.  When I do get a chance to get back to it, I'll likely need to ask for some help from Santiago and Eric, because I'm getting different test failures on my local Mac than on the build machines.  I'm taking that to mean that there are additional dependencies that the build machines have that I am lacking locally.

**Comment by shauheen** [comment]
@montebhoover can you please look into this PR?

**Comment by montebhoover** [comment]
Just an update, some of the tests that were previously failing (TimeSeries.ChangePointDetectionWithSeasonality and two KMeans tests) are now passing.  There is only one test that fails (NormalizerTests.LpGcNormAndWhiteningWorkout) on the VSTS Hosted Mac agents, and there is an open issue where we are working on that: #1506

**Comment by montebhoover** [comment]
FYI @artidoro here's our current progress.
